libyul/optimiser/RedundantAssignEliminator.h

```diff
@@ -164,19 +163,21 @@ class RedundantAssignEliminator: public ASTWalker
 		std::set<YulString> m_outerDeclaredVariables;
 	};
 
+	// TODO check that this does not cause nondeterminism!
+	// This could also be a pseudo-map from state to assignment.
+	using TrackedAssignments = std::map<YulString, std::map<Assignment const*, State>>;
```

Comment:
I'm not sure I find a typedef easier to read than the actual type, but that's my opinion :p

Reply:
But yea, I get that this type is used a lot in the implementation